include/dpp/exception.h

```diff
@@ -53,22 +411,38 @@ class exception : public std::exception
 	 * 
 	 * @param what reason message
 	 */
-	explicit exception(const char* what) : msg(what) { }
+	explicit exception(const char* what) : msg(what), error_code(err_this_is_not_an_error) { }
+
+	/**
+	 * @brief Construct a new exception object
+	 * 
+	 * @param what reason message
+	 * @param code Exception code
+	 */
+	explicit exception(exception_error_code code, const char* what) : msg(what), error_code(code) { }
 
 	/**
 	 * @brief Construct a new exception object
 	 * 
 	 * @param what reason message
 	 * @param len length of reason message
 	 */
-	exception(const char* what, size_t len) : msg(what, len) { }
+	exception(const char* what, size_t len) : msg(what, len), error_code(err_this_is_not_an_error) { }
```

Comment:
Should we make the default "unknown error" or something like that instead? Otherwise this will make the usual way of constructing an exception "not an error", that doesn't feel right

Reply:
we already have unknown error :) its value is 37

Reply:
perhaps `this_exception_has_no_code`